src/modules/app/services/TransactionsService.js

```diff
@@ -0,0 +1,56 @@
+(function () {
+    'use strict';
+
+    /**
+     * @param {app.utils.apiWorker} apiWorker
+     * @param {app.utils.decorators} decorators
+     * @param {User} user
+     * @param {app.utils} utils
+     * @return {AssetsService}
+     */
+    const factory = function (apiWorker, decorators, user, utils) {
+
+        class TransactionsService {
+
+            transactions() {
+                return user.onLogin().then(() => {
+                    return this._getTransactions(user.address);
+                });
+            }
+
+            /**
+             * @param {string} address
+             * @private
+             */
+            @decorators.cachable(2)
+            _getTransactions(address) {
+                return apiWorker.process((Waves, address) => {
+                    return Waves.API.Node.v2.addresses.transactions(address)
+                        .then((list) => {
+                            return list.map((item) => {
```

Comment:
Add a comment about using explicit cloning instead of spread (issue with closure not available in Web Workers).